### PR TITLE
Support multiple YubiKeys per user

### PR DESCRIPTION
**Pull request by J0s3f**
Allow to use multiple yubikeys for the same user.
All IDs can be set in the config.
The changes are compatible with old configs, so no configuration changes are needed, except when adding more ids to a user.

### CODE REVIEW

File: extensions/yubikey/yubikey.extension.php [hunk 1]
```diff
@@ -50,9 +50,9 @@ function store_get_uid_from_yubikey($otp) {
 
 		if (isset($test_user['auth_method']) && $test_user['auth_method'] == 'YUBIKEY') {
 			if (isset($test_user['yubikey']) && is_array($test_user['yubikey'])) {
-				if (isset($test_user['yubikey']['key_id']) && $test_user['yubikey']['key_id'] === $keyID) {
+				if (isset($test_user['yubikey']['key_id']) && in_array($keyID, $test_user['yubikey']['key_id'], true)) {
```
Comment (neverpanic) [inline extensions/yubikey/yubikey.extension.php]: You have a typo in the commit message of this commit. It should be `for each account`, but it `for ach account`.

File: extensions/yubikey/yubikey.extension.php [hunk 2]
```diff
@@ -140,8 +140,11 @@ function yubikey_user_verify_credentials($uid, $credentials) {
 	if ($parts === false) {
 		log_debug('authentication for user ' . $test_user['uid'] . ' failed because the OTP doesn\'t look like one.');
 		return false;
-	}
-	if ($yubi_user['key_id'] !== $parts['prefix']) {
+	}	
+	if(!is_array($yubi_user['key_id'])) {
```
Comment (neverpanic) [inline extensions/yubikey/yubikey.extension.php]: I would prefer if you kept the same formatting rules used in the code already, i.e. a space after `if`.

File: extensions/yubikey/yubikey.extension.php [hunk 3]
```diff
@@ -50,6 +50,9 @@ function store_get_uid_from_yubikey($otp) {
 
 		if (isset($test_user['auth_method']) && $test_user['auth_method'] == 'YUBIKEY') {
 			if (isset($test_user['yubikey']) && is_array($test_user['yubikey'])) {
+				if(isset($test_user['yubikey']['key_id']) && !is_array($test_user['yubikey']['key_id'])) {
```
Comment (neverpanic) [inline extensions/yubikey/yubikey.extension.php]: Space after `if` to match the existing formatting.

**Review by neverpanic — CHANGES_REQUESTED** [review]
Some minor nitpicks, otherwise this looks good to go. Can you force-push over your branch with those fixes?

Thanks for working on this!

**Comment by J0s3f** [comment]
I made the changes you suggested.

**Comment by neverpanic** [comment]
Thanks, merged and released in 1.0.

**Pull request merged**